tools/packaging/packaging.psm1

```
@@ -1229,7 +1229,558 @@ function New-ZipPackage
+</ItemGroup>
+</Project>
+'@
+
+        $nugetConfig =
+@'
+<configuration>
+<packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
+    <add key="powershell-core" value="https://powershell.myget.org/F/powershell-core/api/v3/index.json" />
+</packageSources>
+</configuration>
+'@
+
+        $csproj = $csproj.Replace('!replace_version!', $RefAssemblyVersion).Replace('!replace_snk!', $SnkFilePath)
+        $csproj | Out-File -FilePath "$smaProjectFolder/System.Management.Automation.csproj" -Force
+
+        $nugetConfig | Out-File -FilePath "$genAPIFolder/Nuget.config" -Force
+
+        dotnet build -c Release
+
+        $refBinPath = Join-Path $smaProjectFolder 'bin/Release/netstandard2.0/System.Management.Automation.dll'
+
+        if($refBinPath -eq $null)
+        {
+            throw "Reference assembly was not built."
+        }
+
+        Copy-Item $refBinPath $RefAssemblyDestinationPath -Force
+
+        Write-Verbose -Verbose "Reference assembly built and copied to $RefAssemblyDestinationPath"
+    }
+    finally
+    {
+        Pop-Location
+    }
+
+    if(Test-Path $genAPIFolder)
+    {
+        Remove-Item $genAPIFolder -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    #endregion Build SMA ref assembly
+}
+
 function New-NugetPackage
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $NuSpecPath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string] $PackageDestinationPath
+    )
+
+    $nuget = Get-Command -Type Application nuget.exe -ErrorAction SilentlyContinue
+
+    if($nuget -eq $null)
+    {
+        throw 'nuget.exe is not available in PATH'
+    }
+
+    Push-Location
+    Set-Location $NuSpecPath
+    nuget.exe pack . > $null
```

start-nativeexecution

---

`nuget.exe` indicates this function won't run on Unix, and so are the functions that call this one.
If it's expected, then we should throw an exception on non-windows platforms.

---

Added check for Windows to all functions.